torch/csrc/api/src/optim/lbfgs.cpp

```diff
@@ -30,139 +68,191 @@ Tensor LBFGS::gather_flat_grad() {
       views.push_back(parameter.grad().view(-1));
     }
   }
-  return torch::cat(views);
+  return torch::cat(views, 0);
 }
 
-void LBFGS::add_grad(const torch::Tensor& step_size, const Tensor& update) {
+int64_t LBFGS::_numel() {
+  if(_numel_cache == c10::nullopt) {
+    auto res = 0;
+    for (auto& parameter : _params) {
+      int64_t numel = parameter.numel();
+      res+=numel;
+    }
+    *_numel_cache = res;
+  }
+  return *_numel_cache;
+}
+
+void LBFGS::_add_grad(const double step_size, const Tensor& update) {
   NoGradGuard guard;
```

Comment:
hmm we don't have this in the Python implementation, and can probably remove it here :D

Reply:
weird wonder why only lbfgs doesn't have it. 
cc. @albanD

Reply:
The NoGradGuard in the `step()` function is enough. As this function is called from within the step function.